services/parser-database/parser.py

```diff
@@ -0,0 +1,218 @@
+import logging
+import hashlib
+
+# import datetime
+# import requests
+import json
+
+import slate  # pylint: disable=import-error
+
+import retriever  # pylint: disable=import-error
+# import database  # pylint: disable=import-error
+import keywordmock  # pylint: disable=import-error
+
+logging.basicConfig(level=logging.INFO)
+
+
+class Document:
+    """Class for Regulations Documents"""
+    def __init__(self, jurisdiction, shahash, url, last_updated):
+        self.id = None
+        self.jurisdiction = jurisdiction
+        self.shahash = shahash
+        self.url = url
+        self.last_updated = last_updated
+
+    @staticmethod
+    def from_dict():
+        pass
+
+    def to_dict(self):
+        pass
+
+    def __repr__(self):
+        return f"Document(\
+                id={self.id}, \
+                jurisdiction={self.jurisdiction}, \
+                last_updated={self.last_updated}, \
+                url={self.url}, \
+                shahash={self.shahash}\
+            )"
+
+
+class Article:
+    """Class for storing articles.
+    """
+    def __init__(self, number, text):
+        self.number = number
+        self.text = text
+
+
+articles_in_memory = []
+keywords_in_memory = {}
+
+
+def get_article_by_number(art_num):
+    for item in articles_in_memory:
+        if art_num == str(item["articleNumber"]):
+            return item
+    return None
+
+
+def get_articles_that_match_keywords(keywords_list):
+    """Returns articles that match the the given keyword(s)
+
+    Args:
+        keywords_list (list): Keyword(s) to look for
+
+    Returns:
+        list: articles that match such keyword(s)
+    """
+    to_return = []
+    for keyword in keywords_list:
+        if keyword in keywords_in_memory:
+            print(keyword)
+            to_return.append(keywords_in_memory[keyword])
+    return to_return
+
+
+def count_articles(pdf_text):
+    """Identifies articles and returns a list of Article objects.
+
+    Args:
+        pdf_text (string): contains the PDF text where articles
+        are to be identified.
+
+    Returns:
+        list: article objects
+    """
+    articles = []
+    article_text = ""
+    article_count = 1
+    i = 0
```

Comment:
I would just use a for loop?

Reply:
This generated issues when modifying the `i` value.